src/v/datalake/translation/deps.cc

```diff
@@ -181,9 +181,22 @@ class partition_data_source : public data_source {
     ss::future<kafka::offset> wait_for_data_to_translate(
       std::optional<kafka::offset> last_translated_offset,
       ss::abort_source& as) final {
-        // todo: add logic to wait for enough data to translate.
-        // currently we just break even if a single batch of data is available
-        while (!has_more_data_to_translate(last_translated_offset)) {
+        // naively wait a few polling periods after some data appears, hopefully
+        // grabbing a few more batches before breaking back out to the
+        // translator
+        // TODO: probably choose a better value. configurable? or can we do it
```

Comment:
I was thinking to do it bytes wise (new bytes to be translated since that affects batching) but this is ok too I guess since iceberg/parquet cares about records.

Reply:
the code I wrote here was extremely broken so I dropped it from the PR in the interest of unblocking the next subdivision of remaining work. will revisit in the next PR, which will address other sort of batching oriented stuff.